### PR TITLE
[Tizen] Add code coverage for tests on QEMU

### PR DESCRIPTION
**Pull request by XToripuru**
### Feature

Add code coverage for Tizen tests on QEMU in a similar way other targets do this: `-coverage` target suffix, `LCOV` tool and generating HTML as final output.

This feature uses previously added (#35786) 9P-based file sharing to export `.gcda` files from guest to host and because of that slight path remapping was needed:
```sh
export GCOV_PREFIX=/mnt/chip
export GCOV_PREFIX_STRIP=5
```

As for the integrations tests: `lightning app` was omitted since it is being sandboxed and thus require extra effort.

### Testing

Successfully generated HTML file containing the combined coverage for all* tests.

### CODE REVIEW

**Review by arkq — APPROVED** [review]

File: third_party/tizen/tizen_qemu.py [hunk 1]
```diff
@@ -66,9 +66,10 @@
           "default: $TIZEN_SDK_ROOT/iot-sysdata.img"))
 parser.add_argument(
     '--share', type=str,
-    help=("host directory to share with the guest; if file named 'runner.sh' "
-          "is present at the root of that directory, it will be executed "
-          "automatically after boot"))
+    help=("host directory to share with the guest"))
+parser.add_argument(
+    '--runner', type=str,
+    help=("path to the runner script which will run automatically after boot. ignored in interactive mode. path should be relative to shared directory"))
```
[the remark this answers is not in the document]
Comment (XToripuru) [reply third_party/tizen/tizen_qemu.py]: Done.

File: scripts/build/builders/tizen.py [hunk 2]
```diff
@@ -130,9 +131,55 @@ def __init__(self,
...
+                cmd.append('-j' + str(self.ninja_jobs))
+
+            cmd.append('Tizen')
+
+            self._Execute(cmd, title="Build-only")
+
+            gcov = os.path.join(os.environ['TIZEN_SDK_TOOLCHAIN'], 'bin/arm-linux-gnueabi-gcov')
+
+            self._Execute(['lcov', '--gcov-tool', gcov, '--initial', '--ignore-errors', 'unused', '--capture', '--directory', os.path.join(self.output_dir, 'obj'),
+                           '--exclude', '**/src/controller/*',
```
Comment (andy31415) [inline scripts/build/builders/tizen.py:160]: this should probably be in some separate list as it is repeated and very independent from the other arguments.

Comment (andy31415) [reply scripts/build/builders/tizen.py:160]: It looks like we want some shared gcov arguments and then only have the diffs inside build and post build.

Comment (XToripuru) [reply scripts/build/builders/tizen.py]: Is it okay like this?

File: scripts/build/builders/tizen.py [hunk 3]
```diff
@@ -130,9 +131,59 @@ def __init__(self,
...
+            self._Execute(self.lcov_args() + [
+                '--initial',
+                '--output-file', os.path.join(self.coverage_dir, 'lcov_base.info')
+            ], title="Initial coverage baseline")
+
+    def PostBuildCommand(self):
+        if self.app == TizenApp.TESTS and self.use_coverage:
+
+            self._Execute(self.lcov_args() + ['--output-file', os.path.join(self.coverage_dir,
+                          'lcov_test.info')], title="Initial coverage baseline")
```
Comment (arkq) [inline scripts/build/builders/tizen.py:177]:
```suggestion
                          'lcov_test.info')], title="Update coverage")
```

**Review by andy31415 — APPROVED** [review]

**Pull request merged**